### PR TITLE
Updated release versions for azure-cosmos 4.15.0, azure-spring-data-cosmos 3.7.0 and azure-cosmos-encryption 1.0.0-beta.4

### PR DESCRIPTION
**Pull request by kushagraThapar**
* Updated release versions for azure-cosmos 4.15.0, azure-spring-data-cosmos 3.7.0 and azure-cosmos-encryption 1.0.0-beta.4

#### New Features
* Added `backendLatencyInMs` in `CosmosDiagnostics` for `DIRECT` connection mode.
* Added `retryContext` in `CosmosDiagnostics` for query operations.

#### Key Bug Fixes
* Fixed ignored `HttpClient` decoder configuration issue.
* Fixed incorrect connection mode issue in `CosmosDiagnostics`.
* Fixed issue with handling collisions in the effective partition key.
* Fixed `CosmosQueryRequestOptions` NPE in `readAllItems` API.

#### New Features
* Updated Spring Boot to [spring-boot-dependencies:2.4.5](https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/2.4.5/).
* Updated Spring Cloud to [spring-cloud-dependencies:2020.0.2](https://repo.maven.apache.org/maven2/org/springframework/cloud/spring-cloud-dependencies/2020.0.2/).
* Added `slice` support for queries that do not require page count.
* Updated `azure-cosmos` to version 4.15.0.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-encryption/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # Release History
 
-## 1.0.0-beta.4 (Unreleased)
-
+## 1.0.0-beta.4 (2021-05-12)
+#### New Features
+* Todo
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-encryption/CHANGELOG.md:5]: Add or remove

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Review by moderakh — APPROVED** [review]

File: sdk/cosmos/azure-cosmos-encryption/CHANGELOG.md (hunk 1 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos-encryption/CHANGELOG.md:5]: Yes, we will have one more release before public preview next week, will update the changelog accordingly.

**Pull request merged**